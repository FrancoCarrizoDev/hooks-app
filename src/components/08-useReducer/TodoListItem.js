import React from 'react'

export const TodoListItem = ({ todo, index, handleDeleteTask, handleToggle }) => {
    return (
        <li key={todo.id} className="list-group-item">
            <p
                className={`${todo.done ? 'complete' : ''}`}
                onClick={() => handleToggle(todo.id)}
            >
                {`${index + 1}. ${todo.desc}`}
            </p>
            <button
                className="btn btn-danger"
                onClick={() => handleDeleteTask(todo.id)}
            >
                Borrar
            </button>
        </li>
    )
}
