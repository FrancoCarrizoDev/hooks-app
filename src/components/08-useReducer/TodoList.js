import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({todos, handleDeleteTask, handleToggle}) => {
    return (
        <div>
            <ul className="list-group list-group-flush">
                {todos.map((todo, index) => (
                    <TodoListItem todo={todo} index={index} handleDeleteTask={handleDeleteTask} handleToggle={handleToggle}/>
                ))}
            </ul>
        </div>
    )
}
