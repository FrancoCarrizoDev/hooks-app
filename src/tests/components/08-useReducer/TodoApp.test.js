import { mount, shallow } from "enzyme"
import { demoTodos } from "../../fixtures/demoTodos"
import {TodoApp} from '../../../components/08-useReducer/TodoApp'
import { act } from "@testing-library/react"

describe('should in TodoAppComponent', () => {

    const wrapper = shallow(<TodoApp/>)

    Storage.prototype.setItem = jest.fn(() => {})

    test('should show correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should add an TODO', () => {
        const wrapper = mount(<TodoApp/>)

        act(() => {
            wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[0])
            wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[1])
        })

        expect(wrapper.find('h1').text().trim()).toBe('TodoApp (2)')
        expect(localStorage.setItem).toHaveBeenCalledTimes(2)
    })

    test('should delete an TODO', () => {
        
        wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[0])
        wrapper.find('TodoList').prop('handleDeleteTask')(demoTodos[0].id)

        expect(wrapper.find('h1').text().trim()).toBe('TodoApp (0)')
    })
    
    
    
    
})
