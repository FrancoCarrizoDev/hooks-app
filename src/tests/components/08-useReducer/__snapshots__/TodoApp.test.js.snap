// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should in TodoAppComponent should show correctly 1`] = `
<div
  className="p-5"
>
  <h1>
    TodoApp (
    0
    )
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDeleteTask={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
